R/deployApp.R

```diff
@@ -385,20 +352,46 @@ deployApp <- function(appDir = getwd(),
 
   # invoke post-deploy hook if we have one
   if (deploymentSucceeded) {
-    postDeploy <- getOption("rsconnect.post.deploy")
-    if (is.function(postDeploy)) {
-      if (verbose) {
-        cat("Invoking post-deploy hook rsconnect.post.deploy\n")
-      }
-      postDeploy(appPath)
-    }
+    runDeploymentHook(appPath, "rsconnect.post.deploy", verbose = verbose)
   }
 
   logger("Deployment log finished")
 
   invisible(deploymentSucceeded)
 }
 
+# Shinyapps defaults to public visibility.
+# Other values should be set before data is deployed.
+needsVisibilityChange <- function(server, application, appVisibility = NULL) {
+  if (!isCloudServer(server)) {
+    return(FALSE)
+  }
+  if (is.null(appVisibility)) {
+    return(FALSE)
+  }
+
+  cur <- application$deployment$properties$application.visibility
+  if (is.null(cur)) {
+    cur <- "public"
+  }
+  cur != appVisibility
+}
+
+runDeploymentHook <- function(appPath, option, verbose = FALSE) {
+  hook <- getOption(option)
+  if (!is.function(hook)) {
+    return()
+  }
+
+  name <- gsub("rsconnect.", "", option, fixed = TRUE)
+  name <- gsub(".", "-", name, fixed = TRUE)
+
+  if (verbose) {
+    cat("Invoking ", name, " hook ", option, "\n", sep = "")
```

Comment:
I would be fine simply displaying `Invoking hook: rsconnect.post.deploy` and avoiding the pretty `name`; the hook option is descriptive.